app/scripts/controllers/transactions/index.js

```diff
@@ -336,6 +337,257 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _checkIfTxStatusIsUnapproved(txId) {
+    return (
+      this.txStateManager.getTransaction(txId).status ===
+      TRANSACTION_STATUSES.UNAPPROVED
+    );
+  }
+
+  _updateTransaction(txId, proposedUpdate, note) {
+    const txMeta = this.txStateManager.getTransaction(txId);
+    const updated = merge(txMeta, proposedUpdate);
+    this.txStateManager.updateTransaction(updated, note);
+  }
+
+  /**
+   *
+   * @param {string} txId - transaction id
+   * @param {object} txEIP1559 - holds the eip1559 fees parameters
+   * @param txEIP1559.data
+   * @param txEIP1559.from
+   * @param txEIP1559.to
+   * @param txEIP1559.value
+   * @param txEIP1559.gas
+   */
+  updateEIP1559Params(txId, { data, from, to, value, gas }) {
+    if (!this._checkIfTxStatusIsUnapproved(txId)) {
+      return;
+    }
+
+    const txEIP1559 = {
+      txParams: {
+        data,
+        from,
+        to,
+        value,
+        gas,
+      },
+    };
```

Comment:
None of these are explicitly EIP1559 params so I\m confused by this function name. it seems like this allows for updating of pretty much every txParam field *except* maxBaseFeePerGas and maxPriorityFeePerGas?

Reply:
The function names is because of where the function will be used. So this will be used in 
`ui/ducks/send/send.js:1703` 

```
const eip1559OnlyTxParamsToUpdate = {
        data: txParams.data,
        from: txParams.from,
        to: txParams.to,
        value: txParams.value,
        gas: unapprovedTx.userEditedGasLimit
          ? unapprovedTx.txParams.gas
          : txParams.gas,
      };
      unapprovedTx.originalGasEstimate = eip1559OnlyTxParamsToUpdate.gas;
      const editingTx = {
        ...unapprovedTx,
        txParams: Object.assign(
          unapprovedTx.txParams,
          eip1559support ? eip1559OnlyTxParamsToUpdate : txParams,
        ),
      };
      dispatch(updateTransaction(editingTx));
```

So `dispatch(updateTransaction(editingTx));` will become `dispatch(updateEIP1559Params(editingTx));`

Reply:
`eip1559OnlyTxParamsToUpdate` in the context of https://github.com/MetaMask/metamask-extension/commit/668fd2db660ef43295de74d85034d65c0b14080d should probably be `updateTheseTxParamsIfWeAreOnAnEIP1559Network
`

Reply:
Oh, but by reviewing that commit, I am realizing that this method needs to change a bit...

Reply:
The place where this is meant to be used is https://github.com/MetaMask/metamask-extension/commit/668fd2db660ef43295de74d85034d65c0b14080d#diff-bad76d7279418f3520a0ca93cfaeeb6fcbb1f0d4782f6ff942d422632a1c2780R1521-R1534

For the case where `eip1559support` is false, then that call to `updateTransaction` needs to update more params than this method supports

We need to allow this to update `gasPrice` as well.

Then perhaps we could name it `updateEditableParams`

Reply:
I like updateEditableParams.

Reply:
@danjm any easy way to know the additional params needed when `eip1559support` is false? I see the update function just takes all of the txParams. If all of the txParams should be updated, can we make 2 calls, 

updateEditableParams if `eip1559support` is true
updateTransactionGasFees if false.

Reply:
when it is false, all the same params as what you've currently identified can be updated + only one more: `gasPrice`